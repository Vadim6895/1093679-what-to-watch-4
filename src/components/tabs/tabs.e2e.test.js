import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Tabs from "./tabs.jsx";

Enzyme.configure({
  adapter: new Adapter(),
});

const filmCards = [{
  movieName: `The Grand Budapest Hotel`,
  productionDate: `1984`,
  genre: `Drama`,
  moviePoster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  id: 0,
  director: `Anthony Mann`,
  actors: [`Anthony Mann`],
  rating: 9,
  ratingsQuantity: 250,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.Lorem ipsum dolor sit amet, consectetur adipiscing elit.Lorem ipsum .`,
  length: 120,
  reviews: [{
    text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.Lorem ipsum dolor sit amet, consectetur adipiscing elit.Lorem ipsum .`,
    rating: 8,
    name: `Anthony Mann`,
    date: new Date(),
  }]
}];


it(`Click tabs it correctly`, () => {
  // const mainNavLink = jest.fn();

  const tabs = shallow(
      <Tabs
        activeCard={filmCards[0]}
      />
  );

  const mainNavHandler = tabs.find(`.movie-nav__item`);

  mainNavHandler.forEach((item) => item.simulate(`click`));
  // mainNavHandler.simulate(`click`);
  // tabs.update();
  // expect(mainNavLink.mock.calls.length).toBe(1);
  expect(mainNavHandler.mock.calls.length).toBe(1);
});
