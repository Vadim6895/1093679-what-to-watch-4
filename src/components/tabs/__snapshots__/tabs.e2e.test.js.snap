// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click tabs it correctly 1`] = `ShallowWrapper {}`;
